Android build fails on NotificationExtenderService after upgrading to react-native 0.59.8

**Description:**
I have an application using react-native-onesignal 3.2.14 which was building just fine. After upgrading to react-native 0.59.8 my android build started failing with issues regarding one signal android classes such as `com.onesignal.OSNotificationReceivedResult`, `com.onesignal.NotificationExtenderService` and `com.onesignal.OSNotificationPayload` as I have implemented a custom notification handler to handle silent notifications as described in one signal documentation.
Suddenly after the upgrade none of these classes are found, and trying to figure out where the error was I found the following line:
`OneSignal Warning: Could not get AGP plugin version`

Looking into one-signal-gradle-plugin I found this issue: https://github.com/OneSignal/OneSignal-Gradle-Plugin/issues/68 which may be related, but I did not change or add any plugin to my build.

(your description here)

**Environment**
 I'm using react-native-onesignal 3.2.14 with react-native 0.59.8, react 16.8.3 via *yarn*.


**Steps to Reproduce Issue:**
1. Install the OneSignal SDK using yarn into my project
2. build using react-native 0.57.0 just fine
3. upgrade to react-native 0.59.8.
4. Execute ./gradlew assembleRelease
5. Build fails since it cannot find the referenced classes, with the AGP warning appearing first.

Hi @rgomezp this is the first warning I see when building for android:
`OneSignal Warning: Could not get AGP plugin version`

Then I get this error:

```/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:4: error: cannot find symbol
import com.onesignal.OSNotificationPayload;
                    ^
  symbol:   class OSNotificationPayload
  location: package com.onesignal
/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:5: error: cannot find symbol
import com.onesignal.NotificationExtenderService;
                    ^
  symbol:   class NotificationExtenderService
  location: package com.onesignal
/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:6: error: cannot find symbol
import com.onesignal.OSNotificationReceivedResult;
                    ^
  symbol:   class OSNotificationReceivedResult
  location: package com.onesignal
/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:11: error: cannot find symbol
public class KeenvilNotificationExtender extends NotificationExtenderService {
                                                 ^
  symbol: class NotificationExtenderService
/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:16: error: cannot find symbol
   protected boolean onNotificationProcessing(OSNotificationReceivedResult receivedResult) {
                                              ^
  symbol:   class OSNotificationReceivedResult
  location: class KeenvilNotificationExtender
/Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/services/KeenvilNotificationExtender.java:15: error: method does not override or implement a method from a supertype
   @Override
   ^
Note: /Users/franco/Projects/ironthrone/android/app/src/main/java/com/keenvil/ironthrone/MainApplication.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
6 errors```

Thank you @rgomezp , I'm trying to implement background notification in RN 0.57, but even creating the file "react-native.config.js" on the top of "react-native-onesignal", I'm getting the error to build the project, it says: 

```
error: cannot find symbol 
import com.onesignal.OSNotificationPayload;
```

It's still not working :(
Please help, I am still stuck on and this is unfortunately preventing us from updating our app on the Play Store.
```
error: cannot find symbol
NotificationExtender
```

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -22,15 +22,9 @@ dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
     implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
 
-    implementation('com.onesignal:OneSignal:3.11.1') {
-        // Exclude com.android.support(Android Support library) as the version range starts at 26.0.0
-        //    This is due to compileSdkVersion defaulting to 23 which cant' be lower than the support library version
-        //    And the fact that the default root project is missing the Google Maven repo required to pull down 26.0.0+
-        exclude group: 'com.android.support'
-        // Keeping com.google.android.gms(Google Play services library) as this version range starts at 10.2.1
-    }
-
-    implementation 'com.android.support:cardview-v7:27.1.1'
+    // api is used instead of implementation so the parent :app project can access any of the OneSignal Java
+    //   classes if needed. Such as com.onesignal.NotificationExtenderService
+    api 'com.onesignal:OneSignal:3.11.1'
 
     testImplementation 'junit:junit:4.12'
 }

